src/main/groovy/ui/GroovyMain.java

```diff
@@ -391,17 +391,16 @@ public String getText(String uriOrFilename) throws IOException {
     protected GroovyCodeSource getScriptSource(boolean isScriptFile, String script) throws IOException, URISyntaxException {
         //check the script is currently valid before starting a server against the script
         if (isScriptFile) {
-            if (uriPattern.matcher(script).matches()) {
+            // search for the file and if it exists don't try to use URIs ...
+            File scriptFile = huntForTheScriptFile(script);
+            if (!scriptFile.exists() && uriPattern.matcher(script).matches()) {
                 return new GroovyCodeSource(new URI(script));
-            } else {
-                return new GroovyCodeSource(huntForTheScriptFile(script));
             }
-        } else {
-            return new GroovyCodeSource(script, "script_from_command_line", GroovyShell.DEFAULT_CODE_BASE);
+            return new GroovyCodeSource(huntForTheScriptFile(script));
```

Comment:
can't you just use "scriptFile" instead of the second call to "huntForTheScriptFile"?